lib/internal/fs.js

```diff
@@ -52,7 +53,7 @@ function stringToFlags(flag) {
     case 'xa+': return O_APPEND | O_CREAT | O_RDWR | O_EXCL;
   }
 
-  throw new Error('Unknown file open flag: ' + flag);
+  throw new errors.TypeError('ERR_INVALID_OPT_VALUE', 'flag', flag);
```

Comment:
@gunar Sorry to send you back and forth, I just looked at the docs https://nodejs.org/api/fs.html#fs_fs_open_path_flags_mode_callback
IMHO the formal parameter should be `flags` so for consistency do `/s/flag/flags/` for this whole function.
@joyeecheung do you agree?